packages/devtools_app/integration_test/test_infra/run/run_test.dart

```diff
@@ -273,32 +276,37 @@ class TestRunnerArgs {
     final argWithTestTarget =
         args.firstWhereOrNull((arg) => arg.startsWith(testTargetArg));
     final target = argWithTestTarget?.substring(testTargetArg.length);
-    assert(
-      target != null,
-      'Please specify a test target (e.g. ${testTargetArg}path/to/test.dart',
-    );
```

Comment:
why are we allowing target to be null now?

Reply:
We need to parse the args to see if the `testAppDevice` arg is provided, but the parsing itself was asserting that the `testTarget` wasn't null. Therefore if we were trying to run all web tests (and therefore not specifying a test target), it would fail with the assertion error.

Reply:
Not 100% sure the change I've made is what you asked for, but I added the assertion back and moved the check for whether the test supports the test app device from `main` to `_runTest` (so that we only do the check once we have a test target). LMK what you think!

Reply:
Added a couple suggestions below that would need to be paired with this:
```dart
  if (testTargetProvided) {
    final args = TestRunnerArgs(testRunnerArgs);
    // TODO(kenz): add support for specifying a directory as the target instead
    // of a single file.
    await _runTest(args);
  } else {
    // Run all tests since a target test was not provided.
    final testDirectory = Directory(_testDirectory);
    final testFiles = testDirectory
        .listSync(recursive: true)
        .where((testFile) => testFile.path.endsWith(_testSuffix));
    for (final testFile in testFiles) {
      final testTarget = testFile.path;
      final args = TestRunnerArgs([
        ...testRunnerArgs,
        '${TestRunnerArgs.testTargetArg}$testTarget',
      ]);
      await _runTest(args);
    }
  }
```

Reply:
Oh nice! I think I've applied all the suggestions